armi/physics/neutronics/crossSectionGroupManager.py

```diff
@@ -1069,46 +1083,85 @@ def _setBuGroupBounds(self, upperBuGroupBounds):
                 raise ValueError("Burnup groups must be ascending")
             lastBu = upperBu
 
-    def _updateBurnupGroups(self, blockList):
+        self._buGroupBounds = buGroupBounds + [float("inf")]
+
+    def _setTempGroupBounds(self, tempGroupBounds):
+        """Set the temperature group structure."""
+        lastTemp = -C_TO_K
+        # validate structure
+        for upperTemp in tempGroupBounds:
+            if upperTemp < -C_TO_K:
+                raise ValueError(
+                    "Temperature boundary is below absolute zero {0}.format(upperTemp)"
+                )
+            if upperTemp < lastTemp:
+                raise ValueError("Temp groups must be ascending")
+            lastTemp = upperTemp
+        self._tempGroupBounds = tempGroupBounds + [float("inf")]
+
+    def _updateEnvironmentGroups(self, blockList):
         """
-        Update the burnup group of each block based on its burnup.
+        Update the burnup group of each block based on its burnup and temperature .
 
-        If only one burnup group exists, then this is skipped so as to accomodate the possibility
+        If only one burnup group exists, then this is skipped so as to accommodate the possibility
         of 2-character xsGroup values (useful for detailed V&V models w/o depletion).
 
         See Also
         --------
         armi.reactor.blocks.Block.getMicroSuffix
         """
-        if self._buGroupUpdatesEnabled and len(self._upperBuGroupBounds) > 1:
-            runLog.debug("Updating burnup groups of {0} blocks".format(len(blockList)))
-            for block in blockList:
-                bu = block.p.percentBu
-                for buGroupIndex, upperBu in enumerate(self._upperBuGroupBounds):
-                    if bu <= upperBu:
-                        block.p.buGroupNum = buGroupIndex
-                        break
-                else:
-                    raise ValueError("no bu group found for bu={0}".format(bu))
-        else:
+        if not self._envGroupUpdatesEnabled:
             runLog.debug(
                 "Skipping burnup group update of {0} blocks because it is disabled"
                 "".format(len(blockList))
             )
+            return
+
+        numBuGroups = len(self._buGroupBounds)
+        runLog.debug("Updating env groups of {0} blocks".format(len(blockList)))
+        if len(self._buGroupBounds) == 1 and len(self._tempGroupBounds) == 1:
+            # dont set block.p.envGroupNum since all 1 group and we want to support 2 char xsGroup
+            return
```

Comment:
```suggestion
        if len(self._buGroupBounds) == 1 and len(self._tempGroupBounds) == 1:
            # don't set block.p.envGroupNum since all 1 group and we want to support 2 char xsGroup
            return
        runLog.debug(f"Updating env groups of {len(blockList))} blocks"
```

Reply:
reordered.